beamer/state_machine.py

```diff
@@ -278,13 +279,16 @@ def _handle_initiate_l1_resolution(
 
     assert request.fill_tx is not None, "Request not yet filled"
     if _l1_resolution_criteria_fulfilled(claim, context):
-        request.l1_resolution_tx_handle = context.resolution_pool.submit(
+        future = context.resolution_pool.submit(
             run_relayer,
             context.config.l1_rpc_url,
             context.config.l2b_rpc_url,
             context.config.account.key,
             request.fill_tx,
         )
+        context.l1_resolutions[request.id] = future
+        request.l1_resolution_started = True
```

Comment:
We could consider using the future's `add_done_callback` here so that we don't need a separate `process_l1_resolutions` function -- the futures can remove themselves from the dict once they are done.

Reply:
That version is implemented in bc111f054b70014fdc8b4a4f33a84e434d3cfcf7

But I'm not sure if I prefer it over the other, more explicit one. You're choice.

Reply:
The one difference that comes to my mind is that the latter might result in removal of the future from the dict somewhat earlier than the former, due to the former being tied to the loop processing. But I don't think we should care too much about that, so I'm also fine with your original approach.